src/UI/Art/SpriteArtist/PlayerSetupArtist.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    // Define the space to draw the list of player CO portraits.
+    int playerXCenter = ((imageWidth - readyAreaWidth) / 2);
+    int highlightedPlayerYCenter = imageHeight / 2; // Whichever player has focus should be centered.
+
+    // Find where the zeroth player CO should be drawn.
+    // Shift from the center location by the spacing times the number of the highlighted option.
+    int playerYCenter = (int)(highlightedPlayerYCenter - (animHighlightedPlayer * panelHeight));
+
+    // Draw all of the player info.
+    for(int i = 0; i < numCOs; ++i, playerYCenter += (panelHeight))
+    {
+      // Only bother to draw it if it is on-screen.
+      if( (playerYCenter > -panelHeight/2) && ( playerYCenter < SpriteOptions.getScreenDimensions().getHeight()+(panelHeight/2) ) )
+      {
+        PlayerSetupInfo playerInfo = myControl.getPlayerInfo(i);
+        Integer key = new Integer(i);
+
+        // Get the relevant PlayerPanel.
+        if( !playerPanels.containsKey(key) ) playerPanels.put(key, new PlayerPanel(playerInfo));
+        PlayerPanel panel = playerPanels.get(key);
+
+        // Update the PlayerPanel and render it to an image.
+        BufferedImage playerImage = panel.update(playerInfo);
+
+        int drawX = playerXCenter - (playerImage.getWidth())/2;
+        int drawY = playerYCenter - (playerImage.getHeight())/2;
+        myG.drawImage(playerImage, drawX, drawY, playerImage.getWidth(), playerImage.getHeight(), null);
+      }
+    }
+
+    // Figure out where to draw the cursor.
+    if( myControl.getHighlightedCategory() == PlayerSetupController.SelectionCategories.START.ordinal() )
+    {
+      // Ready is currently selected.
+      SpriteUIUtils.drawCursor(myG, readyX, readyY, readyButton.getWidth(), readyButton.getHeight(), myControl.getPlayerInfo(highlightedPlayer).getCurrentColor());
+    }
+    else // Draw the cursor over the appropriate player info panel.
+    {
+      PlayerPanel panel = playerPanels.get(highlightedPlayer);
+      SpriteUIUtils.ImageFrame pane;
+      switch(myControl.getHighlightedCategory())
+      {
+        default:
+        case 0: // Commander
+          pane = panel.commanderPane;
+          break;
+        case 1: // Faction/Color
+          pane = panel.unitPane;
+          break;
+        case 2: // Team
+          pane = panel.teamPane;
+          break;
+        case 3: // AI Controller
+          pane = panel.aiPane;
+          break;
+      }
+      PlayerSetupInfo info = myControl.getPlayerInfo(highlightedPlayer);
+      BufferedImage playerImage = panel.getImage();
+      int drawX = playerXCenter - (playerImage.getWidth())/2;
+      int drawY = imageHeight / 2 - playerImage.getHeight()/2;
+      SpriteUIUtils.drawCursor(myG, drawX+pane.xPos, drawY+pane.yPos, pane.width, pane.height, info.getCurrentColor());
+    }
+
+    // Finally, draw our rendered image onto the window.
+    g.drawImage(image, 0, 0, dimensions.width, dimensions.height, null);
```

Comment:
I'm not especially a fan of this, combined with `X = dimensions.X / drawScale`.
It works, sure, but it can distort stuff and it's pretty obvious as you resize the window.

Reply:
Distortion goes away if you replace it with:
`g.drawImage(image, 0, 0, imageWidth*drawScale, imageHeight*drawScale, null);`

Reply:
Hm. It does distort things a bit, doesn't it? I'll take your fix. I think technically that could leave us up to `drawScale` pixels shy of the window's edge, but we don't actually draw this stuff to the window's edge.